e2e/tests/omnibus-arbitrum-mainnet.toml/Liquidation_Rewards_Registrations.e2e.js

```diff
@@ -0,0 +1,170 @@
+  // const accountId = 1337;
+  const address = wallet.address;
+  const privateKey = wallet.privateKey;
+  const poolId = '1';
+
+  let registeredDistributors;
+  let liquidationDistributors;
+  let expectedDistributors;
+
+  let snapshot;
+  before('Create snapshot', async () => {
+    snapshot = await provider.send('evm_snapshot', []);
+    log('Create snapshot', { snapshot });
+  });
+  after('Restore snapshot', async () => {
+    log('Restore snapshot', { snapshot });
+    await provider.send('evm_revert', [snapshot]);
+  });
+
+  it('should sync time of the fork', async () => {
+    await syncTime();
+  });
+
+  it('should collect all registered rewards distributors', async () => {
+    const cannon = require('../../deployments/cannon.json');
+
+    registeredDistributors = [];
+    // walk over all the registration invokes and set isRegistered flag for deployed distributors
+    for (const [key, value] of Object.entries(cannon?.state || {})) {
+      if (key.startsWith('invoke.')) {
+        const [, artifactName] = key.split('.');
+        const events = value?.artifacts?.txns?.[artifactName]?.events?.RewardsDistributorRegistered;
+        if (events) {
+          for (const event of events) {
+            const [poolId, collateralType, distributor] = event.args;
+            registeredDistributors.push({ poolId, collateralType, distributor });
+          }
+        }
+      }
+    }
+    log({ registeredDistributors });
+  });
+
+  it('should collect all expected distributors for perps liquidations', async () => {
+    const supportedCollateralsIds = await contractRead({
+      wallet,
+      contract: 'PerpsMarketProxy',
+      func: 'getSupportedCollaterals',
+      args: [],
+    });
+    log({ supportedCollateralsIds });
+
+    const systemToken = require('../../deployments/systemToken.json');
+    const synthTokens = require('../../deployments/synthTokens.json');
+
+    // We can only use synth tokens and system token as collaterals
+    // We do NOT need a rewards distributor for system token
+    const supportedCollaterals = supportedCollateralsIds
+      .map((id) => {
+        const synthMarketId = id.toString();
+        if (synthMarketId === '0') {
+          return {
+            synthMarketId,
+            ...systemToken,
+          };
+        }
+        return synthTokens.find((token) => token.synthMarketId === synthMarketId);
+      })
+      .filter(Boolean);
+    log({ supportedCollaterals });
+
+    const liquidatableCollaterals = supportedCollaterals
+      // We do not need rewards distributor for system token
+      .filter((collateral) => collateral.address !== systemToken.address);
+
+    expectedDistributors = [];
+    for (const liquidatableCollateral of liquidatableCollaterals) {
+      const { distributor, poolDelegatedCollateralTypes } = await contractRead({
+        wallet,
+        contract: 'PerpsMarketProxy',
+        func: 'getRegisteredDistributor',
```

Comment:
Each registered liquidation rewards distributor has one distributor address and a list of pool collaterals

As configured in this block

```toml
[invoke.PerpsMarketProxy_registerDistributor_sBTC_liquidations]
target = ["perpsFactory.PerpsMarketProxy"]
fromCall.func = "owner"
func = "registerDistributor"
args = [
    "<%= extras.synth_btc_token_address %>",
    "<%= imports.RewardsDistributor_sBTC_liquidation_rewards.contracts.RewardsDistributorExternal.address %>",
    "<%= extras.synth_btc_market_id %>",
    [
        "<%= settings.arb_address %>",
        "<%= settings.usdc_address %>",
        "<%= settings.usde_address %>",
        "<%= settings.weth_address %>",
        "<%= settings.weeth_address %>",
        "<%= settings.wsteth_address %>",
    ],
]
```

So given we have 6 pool collaterals and 4 perps collaterals we end up with a list of 24 expected distributors

Each distributor does not have to be a separate contract deployed per collateral and we use one single pool level distributor. But this pool level distributor must be registered per each pool collateral.

So we end up with 4 Rewards Distributor contracts (based on a payout token, one per each  perps collateral: stBTC, sETH, swSOL, sUSDe). But each of them is also registered 6 times separately for each of the pool collaterals (arb, usdc, usde, weth, weeth, wsteth)

This could have been avoided if Perps market had support for pool-level distributors and did not iterate over each pool collateral itself